### ISSUE TITLE
XMLHttpRequest on the main thread is deprecated, any way to update ?

### ISSUE THREAD

**Issue by afidegnum** [brython-dev/brython#520]
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/

this is what i am facing while trying the ajax functionality.

**Comment by kikocorreoso** [comment]
This is a minimum example of an *async* ajax request:

```
from browser import ajax, alert

def on_complete(req):
    if req.status==200 or req.status==0:
        alert(req.text)
    else:
        alert( "error "+req.text)

# AJAX request to https://jsonplaceholder.typicode.com/
req = ajax.ajax()

url = "https://jsonplaceholder.typicode.com/posts/1"
method = 'GET'
async = True
req.bind('complete',on_complete)
req.open(method, url, async)
req.send()
```

You can test it here: http://brython.info/tests/editor.html?lang=en

If you are seeing the `XMLHttpRequest on the main thread is deprecated` message maybe it is related with the import machinery (see [here](https://github.com/brython-dev/brython/blob/ff00c0a039e10c44225fa10776d30c00f87f1e5b/www/src/py_import.js#L77)). 

Just post some code as it will simpler to provide a more accurate answer.

**Comment by afidegnum** [comment]
Well, i m facing an imminent problem

my ajax call return an entire json structure, and i m having hard time using ajax to fetch elements i need. when i tried to import urllib2 or another module, i faced errors, 

I m trying to fetch json data into the option element of select list, 

here is an example of what im trying to fetch

`http://gsam.ga/mnews`


What's the best way to retain specific element in the json try to be used in other locations?

how do i import additional modules?

**Comment by kikocorreoso** [comment]
Could you post your complete code to better understand what you want to do?

**Comment by afidegnum** [comment]
sorry for the late reply, 

this code neads to _n_ not defined error, 

` {% extends "gsam/innerbase.html" %}
{% block title %} {{ super() }}Project Management {% endblock %}
{% block headscript %} {{ super() }}     <script type="text/javascript" src="{{ url_for('static', filename='Brython/brython.js') }}"></script>  {% endblock %}

  {% block body_attribs %}  onload="brython(1)" {% endblock body_attribs %}>

            {% block rowblock %}



                            <div class="row">
  {% block coldiv -%}<div class="content col-sm-8 col-md-9">  {% endblock coldiv -%}
                        <div class="posts">


<div id="result">(empty)</div>



<script type="text/python">

from browser import document, alert, html
container = document['formblock']
newdiv = html.DIV(id = "new-div")
newdiv.style = {"padding": "5px", 
               "backgroundColor": "#ADD8E6"}


from browser import document as doc
from browser import html, alert, ajax

def update_select(ev):
	rank = choices.index(ev.target.value)
	sel.options[rank].selected

def show_selected(ev):
    alert([option.value for option in sel if option.selected])

    

timeout = 4

def fake_qs():
    return '' %time.time()

def go(url):
    req = ajax.ajax()
    req.bind('complete',on_complete)
    req.set_timeout(timeout,err_msg)
    req.open('GET',url+fake_qs(),True)
    req.send({'articles':34})

def post(url):
    req = ajax.ajax()
    req.bind('complete',on_complete)
    req.set_timeout(timeout,err_msg)
    req.open('POST',url,True)
    req.set_header('content-type','application/x-www-form-urlencoded')
    req.send({'foo':34})

list = go('http://136.243.58.29/mnews')



doc['panel'] <= sel

b_show = html.BUTTON('show selected')
b_show.bind('click', show_selected)
doc['panel'] <= b_show

form = html.FORM()
input1 = html.INPUT(type="text", name="firstname", value="First name")
input2 = html.INPUT(type="text", name="lastname", value="Last name")

sel = html.SELECT(size=len(list), multiple=True)
for item in list['article']:
    sel <= html.OPTION(item['id'])

input3 = html.BUTTON("Button with no action!")
form <= input1 + html.BR() + input2 + html.BR() + sel + input3

newdiv <= form + html.BR()

container <= newdiv






</script>







    </div>

    <div class="center">

</div>
                </div>

                <div class="sidebar col-sm-4 col-md-3">
                        <div class="widget">
    <div class="widget-title">
        <h2>Custom Menu</h2>
    </div>

    <div class="widget-content">
        <ul class="menu">
            <li><a href="blog-index.html#">Home</a></li>
        </ul>
    </div>
</div>

    <div class="widget">
    <div class="widget-title">
        <h2>Categories</h2>
    </div>


</div>

    <div class="widget">



</div>
                </div>
            </div>
            {% endblock rowblock %}`



here is the code i m working on...

**Comment by kikocorreoso** [comment]
There are several stuff in your code that are not correct.

* You are using `time.time()` but it is not imported before.
* You are using the `go` function to get the list of posts but in the `go` function you are using a url that is not correct (`url+fake_qs()`). 
* You are not defining the `on_complete` callback used in the `go`.
* There is a lot of code that is not useful to help you debug the problem
* ...

It seems your problem is not related with a brython bug so it is best to provide a clean minimum example that reproduces your error in the [mail list](https://groups.google.com/forum/?fromgroups=#!forum/brython). 

Let's continue this discussion on the mail list.

I close the issue now and if we can find a bug from the discussion on the mail list we can re-open this issue or we can open a new one.
